bundles/org.openhab.binding.haywardomnilogic/src/main/resources/OH-INF/thing/pump.xml

```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="haywardomnilogic"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="pump" listed="false">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+
+		<label>Pump</label>
+		<description>Pump</description>
+		<channels>
+			<channel id="pumpEnable" typeId="system.power"/>
+			<channel id="pumpSpeed" typeId="pumpSpeed"/>
+		</channels>
+
+		<properties>
+			<property name="vendor">Hayward</property>
+			<property name="minPumpSpeed"></property>
+			<property name="maxPumpSPeed"></property>
+			<property name="minPumpRPM"></property>
+			<property name="maxPumpRPM"></property>
+		</properties>
+		<representation-property>systemID</representation-property>
+
+	</thing-type>
+
+	<channel-type id="pumpSpeed">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Pump Speed</label>
+		<description>Pump Speed</description>
+		<state min="0" max="100" step="1.0" pattern="%1f" readOnly="false"/>
```

Comment:
The pump and filter accept a speed value from 0 to 100.  It seems like unnecessary overhead/math to get the Dimmer 0.00% out of the DecimalType, convert it to a float, multiply x100, converter to integer and then to a string, but I can do that if you think its necessary.  If you have any hints on shortening up this operation and stepping through the datatypes, please let me know.

Reply:
Value conversion aside, I was suggesting `Dimmer` due to the additional commands it can support: Increase,Decrease, On and Off. If support of those commands doesn't make sense for this channel then by all means keep it as `Number:Dimensionless`.

Reply:
Also, to answer your question, the Dimmer's `PercentType` extends DecimalType and stores its value as a number between 0 and 100. You don't need to do any multiplication to convert it to an integer, calling `.intValue()` should suffice.